### PR TITLE
feat: add semantic-release setup for automatic deployments

### PR DESCRIPTION
**Pull request by matmalkowski**
This adds the setup for semantic-release. master/beta/alpha branch changes will get deployed to npm
with version depending on the commit message

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+on:
+    push:
+        branches:
+            - master
+            - alpha
+            - beta
+jobs:
+    release:
+        name: Release
+        runs-on: ubuntu-18.04
```
Comment (victorhqc) [inline .github/workflows/release.yml:11]: There's also the `ubuntu-latest` option

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        steps:
+            - name: Checkout
+              uses: actions/checkout@v2
+              with:
+                  fetch-depth: 0
+            - name: Setup Node.js
+              uses: actions/setup-node@v1
+              with:
+                  node-version: 12
```
Comment (victorhqc) [inline .github/workflows/release.yml:20]: 👍

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+            - name: Install dependencies
+              run: yarn install --frozen-lockfile
+            - name: Build the package
+              run: yarn build
+            - name: Release
+              env:
+                  GH_TOKEN: ${{ secrets.GH_TOKEN }}
+                  NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+              run: npx semantic-release
```
Comment (victorhqc) [inline .github/workflows/release.yml:29]: cool

**Review by victorhqc — APPROVED** [review]
looks good to me

**Comment by hardchor** [comment]
Might be worth adding a note to the contribution guide what is expected from the commit message to get picked up by this?

**Comment by matmalkowski** [comment]
@hardchor Yeah, I will make sure it makes it to the `Contributing Guide` once we have that doc for v2 👍🏻

**Pull request merged**